### PR TITLE
set podName by Downward API && log podName

### PR DESCRIPTION
**Pull request by theonefx**
# Description
Set POD_NAME by Downward API when inject Dapr sidecar.
Log sidecar pod name in Dapr Operator.

## Issue reference

Please reference the issue this PR will close: https://github.com/dapr/dapr/issues/3947

## Checklist

Please make sure you've  completed the relevant tasks for this PR, out of the following list:

* [X] Code compiles correctly
* [X] Created/updated tests
* [X] Unit tests passing
* [x] End-to-end tests passing
* [ ] Extended the documentation / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Specification has been updated / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_
* [ ] Provided sample for the feature / Created issue in the https://github.com/dapr/docs/ repo: dapr/docs#_[issue number]_

### CODE REVIEW

**Comment by theonefx** [comment]
@yaron2

File: pkg/components/kubernetes_loader.go [hunk 1]
```diff
@@ -39,21 +39,24 @@ type KubernetesComponents struct {
...
 }
 
 // NewKubernetesComponents returns a new kubernetes loader.
-func NewKubernetesComponents(configuration config.KubernetesConfig, namespace string, operatorClient operatorv1pb.OperatorClient) *KubernetesComponents {
+func NewKubernetesComponents(configuration config.KubernetesConfig, namespace string, operatorClient operatorv1pb.OperatorClient, name string) *KubernetesComponents {
 	return &KubernetesComponents{
 		config:    configuration,
 		client:    operatorClient,
 		namespace: namespace,
+		podName:   name,
```
Comment (daixiang0) [inline pkg/components/kubernetes_loader.go:51]: Please add tests.

Comment (theonefx) [reply pkg/components/kubernetes_loader.go:51]: done

**Review by daixiang0 — DISMISSED** [review]
LGTM overall, let's wait the test results.

File: pkg/operator/api/api.go [hunk 2]
```diff
@@ -134,13 +134,13 @@ func (a *apiServer) ListComponents(ctx context.Context, in *operatorv1pb.ListCom
...
 		if err != nil {
-			log.Warnf("error processing component %s secrets: %s", c.Name, err)
+			log.Warnf("error processing component %s secrets from pod %s/%s: %s", c.Name, in.Namespace, in.PodName, err)
 			return &operatorv1pb.ListComponentResponse{}, err
 		}
 
 		b, err := json.Marshal(&c)
 		if err != nil {
-			log.Warnf("error marshalling component %s : %s", c.Name, err)
+			log.Warnf("error marshalling component %s from pod %s/%s: %s", c.Name, in.Namespace, in.PodName, err)
```
Comment (beiwei30) [inline pkg/operator/api/api.go:143]: pls. add more tests to increase coverage.

File: pkg/operator/api/api.go [hunk 3]
```diff
@@ -134,13 +134,13 @@ func (a *apiServer) ListComponents(ctx context.Context, in *operatorv1pb.ListCom
 		c := components.Items[i] // Make a copy since we will refer to this as a reference in this loop.
```
Comment (beiwei30) [inline pkg/operator/api/api.go:134]: The current change make sure the pod info exposed for `ListComponents` and `ComponentUpdate` only, we should make sure the same info should be exposed for `GetConfiguration` and `ListSubscriptions`

File: pkg/components/kubernetes_loader.go [hunk 4]
```diff
@@ -39,21 +39,24 @@ type KubernetesComponents struct {
 	config    config.KubernetesConfig
 	client    operatorv1pb.OperatorClient
 	namespace string
+	podName   string
 }
 
 // NewKubernetesComponents returns a new kubernetes loader.
-func NewKubernetesComponents(configuration config.KubernetesConfig, namespace string, operatorClient operatorv1pb.OperatorClient) *KubernetesComponents {
+func NewKubernetesComponents(configuration config.KubernetesConfig, namespace string, operatorClient operatorv1pb.OperatorClient, name string) *KubernetesComponents {
```
Comment (beiwei30) [inline pkg/components/kubernetes_loader.go:46]: rename `name` to `podName`?

**Review by beiwei30 — CHANGES_REQUESTED** [review]

File: pkg/operator/api/api.go (hunk 3 above)
Comment (theonefx) [reply pkg/operator/api/api.go:133]: done，all 4 method now support pod info

File: pkg/components/kubernetes_loader.go (hunk 4 above)
Comment (theonefx) [reply pkg/components/kubernetes_loader.go]: done

**Review by beiwei30 — DISMISSED** [review]
LGTM

**Comment by yaron2** [comment]
@theonefx this is a great PR, thanks. there's only one problem - because the proto methods for the operator changed, this will be a breaking change. So in an upgrade scenario to 1.6, if we upgrade the control plane and any of the existing sidecars restart, they'll be crashing.

So my suggestion is to create a `ListSubsciptionsV2` method and have the sidecar consume that one, then in 2 releases we can remove the old method.

**Review by artursouza — DISMISSED** [review]
Breaking changes as per Yaron's point.

**Comment by theonefx** [comment]
@yaron2 
In my new change, `operator` support both `ListSubscriptions` and `ListSubscriptionsV2` methods. And the new daprd use `V2` method to expose pod info.

**Comment by theonefx** [comment]
@artursouza

**Comment by yaron2** [comment]
Thanks @theonefx will take a look.

**Comment by theonefx** [comment]
@yaron2 @artursouza Is there any new progress?

**Review by yaron2 — APPROVED** [review]
LGTM

**Comment by yaron2** [comment]
Merged, thanks @theonefx !

**Pull request merged**